watidl/wasi-core.watidl

```diff
@@ -0,0 +1,1076 @@
+(type $size_t u64)
```

Comment:
This is interesting :-). I assume the motivation for using 64-bit here is to be generic between 32-bit and 64-bit. I think that's reasonable, especially since most uses of `size_t` in `wasi_unstable` are replaced here by bindings types like string and array. However, it'd be good to add a comment about this here.